src/main/java/frc/robot/subsystems/swerve/ModuleIOReal.java

```diff
@@ -145,23 +144,22 @@ public ModuleIOReal(ModuleConstants constants) {
     cancoderConfig.MagnetSensor.MagnetOffset = constants.cancoderOffset().getRotations();
     cancoder.getConfigurator().apply(cancoderConfig);
 
-    timestampQueue = PhoenixOdometryThread.getInstance().makeTimestampQueue();
-
     drivePosition = driveTalon.getPosition();
-    drivePositionQueue =
-        PhoenixOdometryThread.getInstance().registerSignal(driveTalon, driveTalon.getPosition());
     driveVelocity = driveTalon.getVelocity();
     driveAppliedVolts = driveTalon.getMotorVoltage();
     driveCurrent = driveTalon.getStatorCurrent();
 
     turnAbsolutePosition = cancoder.getAbsolutePosition();
     turnPosition = turnTalon.getPosition();
-    turnPositionQueue =
-        PhoenixOdometryThread.getInstance().registerSignal(turnTalon, turnTalon.getPosition());
     turnVelocity = turnTalon.getVelocity();
     turnAppliedVolts = turnTalon.getMotorVoltage();
     turnCurrent = turnTalon.getStatorCurrent();
 
+    PhoenixOdometryThread.getInstance()
+        .registerSignals(
+            new Registration(driveTalon, ImmutableSet.of(drivePosition)),
+            new Registration(turnTalon, ImmutableSet.of(turnPosition)));
```

Comment:
So, I knew I wanted the api to support registering multiple signals to one parent device at once, so options are some time kind of [Collection](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html) or an Array. Arrays are pretty awkward to work with in Java most of the time, so unless I've got a specific reason to use them I don't by default. Of the major `Collection` types we're looking at `Set` and `List` since we don't need something with more complicated semantics like a `Queue` or a `Stack`. The major difference between `Set`s and `List`s is that `Set`s have unique elements while `List`s don't and `List`s have an order to them while `Set`s don't. The signals we're registering should be unique (no reason to register any of them more than once - in fact we should avoid this) and don't have an order. So `Set` it is. I'm passing data that shouldn't need to be modified (and won't stick around very long) so `ImmutableSet` is a reasonable choice. I could have as easily used `HashSet` - the signature for `registerSignals` only requires a `Set`, not a particular implementation of one.

Reply:
that makes sense, I was mostly wondering why we need to support multiple signals per device since I don't believe it's ever implemented with multiple signals and i imagine that could cause issues?